File: java/palindromenumber.java
public class Solution {
    public boolean isPalindrome(int x) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(x<0) return false;
        if(x<10) return true;
        int k = 1;
        int kx = x;
        while(true){
            if(kx<10){
                break;
            }else{
                kx/=10;
                k*=10;
            }
        }
        
        
        while(x>=10){
            int left = x/k;
            int right = x%10;
            if(left!=right){
                return false;
            }
            x = x%k;
            x = x/10;
            k/=100;
        }
        return true;
    }
}

File: PalindromeNumber.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) {
            return false;
        }
        int start = x;
        int startMultiplier = 1;
        while(start>=10) {
            start = start/10;
            startMultiplier *=10;
        }
        while(startMultiplier>0) {
            int end = x%10;
            start = x/startMultiplier;
            if(start!=end) {
                return false;
            }
            x -= (start*startMultiplier+end);
            startMultiplier /= 100;
            x /= 10;
        }
        return true;
    }
};
